src/user-guide/writing-workflows/scheduling.rst

```diff
@@ -1528,29 +1536,31 @@ in :ref:`Section External Triggers`.
 
 .. _User Guide Required Outputs:
 .. _required outputs:
-.. _incomplete tasks:
 
 Required Outputs
 ----------------
 
 .. versionadded:: 8.0.0
 
-:term:`Task outputs <task output>` in the :term:`graph` are either
-:term:`required <required output>` (the default) or  :term:`optional <optional
-output>`.
+:term:`Task outputs <task output>` in the :term:`graph` can be
+:term:`required <required output>` (the default) or
+:term:`optional <optional output>` (marked with ``?`` in the graph).
 
-The scheduler requires all task outputs to be completed at runtime, unless they
-are marked with ``?`` as optional. This allows it to correctly diagnose
+Tasks are expected to complete required outputs at runtime, but
+they don't have to complete optional outputs.
+
+This allows the scheduler to correctly diagnose
 :term:`workflow completion`. [2]_
 
-Tasks that finish without completing required outputs  [3]_ are retained as
-:ref:`incomplete <incomplete tasks>` pending user intervention, e.g. to be
-retriggered after a bug fix.
+Tasks that achieve a :term:`final status` without completing their
+outputs [3]_ are retained in the :term:`n=0 window <n-window>` pending user
+intervention, e.g. to be retriggered after a bug fix.
 
 .. note::
-   Incomplete tasks stall the workflow if there is nothing else to do (see
-   :ref:`workflow completion`). They also count toward the :term:`runahead
-   limit`, because they may run again once dealt with.
+   Tasks that achieve a final status without completing their outputs will
+   raise a warning and stall the workflow when there is nothing else for
+   the scheduler to run (see :ref:`workflow completion`). They also count
+   toward the :term:`runahead limit`, because they may run again once dealt with.
```

Comment:
> because they may run again once dealt with.

???

```suggestion
   toward the :term:`runahead limit`.
```

Reply:
The point is, even though they have final status, they are not done with as far as the workflow is concerned and may well get run again by manual intervention - which explains why they should count toward the runahead limit.

However, I agree that's not a very elegant way of putting it!